app/src/main/java/com/android/unio/model/association/AssociationRepositoryFirestore.kt

```diff
@@ -72,11 +74,17 @@ class AssociationRepositoryFirestore @Inject constructor(private val db: Firebas
       onSuccess: (Association) -> Unit,
       onFailure: (Exception) -> Unit
   ) {
-    getAssociationRef(id)
-        .get()
-        .performFirestoreOperation(
-            onSuccess = { document -> onSuccess(hydrate(document.data)) },
-            onFailure = { exception -> onFailure(exception) })
+    getAssociationRef(id).registerSnapshotListener(MetadataChanges.EXCLUDE) {
+        documentSnapshot,
+        exception ->
+      if (exception != null) {
+        onFailure(exception)
+        return@registerSnapshotListener
+      }
+      if (documentSnapshot != null && documentSnapshot.exists()) {
+        onSuccess(hydrate(documentSnapshot.data))
+      }
+    }
```

Comment:
As I didn't work on the offline mode, I'm not sure I understand what this does or what the fix is? Could you maybe comment this some more?

Reply:
Check #235 and #228 for a description of the issue. Here, instead of using success and failure listener directly, we use a Snapshot Listener that call directly the callback when a read/write are made on the local (cache) database. This prevent this feature to be stuck for a long time without result. See [this](https://firebase.google.com/docs/firestore/query-data/listen?hl=fr) for more info

Reply:
Ok thanks!